ray-operator/apis/ray/v1/rayjob_types.go

```diff
@@ -64,6 +64,10 @@ type RayJobSpec struct {
 	RayClusterSpec *RayClusterSpec `json:"rayClusterSpec,omitempty"`
 	// clusterSelector is used to select running rayclusters by labels
 	ClusterSelector map[string]string `json:"clusterSelector,omitempty"`
+	// If LightWeightSubmissionMode is true, KubeRay operator sends a request to the RayCluster to create a
+	// Ray job. Otherwise, the operator creates a submitter Kubernetes Job to submit the Ray job.
+	// +kubebuilder:default:=false
+	LightWeightSubmissionMode bool `json:"lightWeightSubmissionMode,omitempty"`
```

Comment:
I wonder if it's cleaner to enumerate the submission type instead of a bool.
```
type RayJobSpec struct {
...
...

SubmissionMode JobSubmissionMode  
}
```

type JobSubmissionMode string

const (
    // Submit job via Kubernetes Job
    Job JobSubmissionMode = "batchv1.Job"
    // Submit job via HTTP request
    HTTP JobSubmissionMode = "HTTP"
    ...
)

Reply:
I guess it depends on whether we anticipate more modes of job submission going forward.

Reply:
Good idea. Updated.